File: MyHashMap.py
class MyHashMap(object):

    def __init__(self):
        self.size = 1000
        self.bucket = [[] for _ in range(self.size)]

        
    def _has(self,key):
        return key % self.size
    def put(self, key, value):
        """
        :type key: int
        :type value: int
        :rtype: None
        """
        index = self._has(key)
        for i,(k,v) in enumerate (self.bucket[index]):
            if k == key:
                self.bucket[index][i] =(key, value)
                return
        self.bucket[index].append((key,value))

        

    def get(self, key):
        """
        :type key: int
        :rtype: int
        """
        index = self._has(key)
        for k,v in self.bucket[index]:
            if k == key:
                return v 
        return -1

        

    def remove(self, key):
        """
        :type key: int
        :rtype: None
        """
        index = self._has(key)
        for i, (k,v) in enumerate (self.bucket[index]):
            if k ==key:
                del self.bucket[index][i]
                return

        


# Your MyHashMap object will be instantiated and called as such:
# obj = MyHashMap()
# obj.put(key,value)
# param_2 = obj.get(key)
# obj.remove(key)

File: MyHashMap.cs
public class MyHashMap
{

    int[][] hashMap;
    int bucket, bucketItem;
    public MyHashMap()
    {
        this.bucket = 1000;
        this.bucketItem = 1000;
        hashMap = new int[this.bucket][];
    }

    public int GetBucket(int key)
    {
        return key % this.bucket;
    }

    public int GetBucketItem(int key)
    {
        return key / this.bucketItem;
    }

    public void Put(int key, int value)
    {
        var bucket = GetBucket(key);
        var bucketItem = GetBucketItem(key);
        if (hashMap[bucket] == null)
        {
            var bucketItemCount = bucket == 0 ? this.bucketItem + 1 : this.bucketItem;
            hashMap[bucket] = new int[bucketItemCount];
            for (var i = 0; i < bucketItemCount; i++)
            {
                hashMap[bucket][i] = -1;
            }
        }
        hashMap[bucket][bucketItem] = value;
    }

    public int Get(int key)
    {
        var bucket = GetBucket(key);
        var bucketItem = GetBucketItem(key);
        if (hashMap[bucket] == null)
        {
            return -1;
        }
        return hashMap[bucket][bucketItem];

    }

    public void Remove(int key)
    {
        var bucket = GetBucket(key);
        var bucketItem = GetBucketItem(key);
        if (hashMap[bucket] == null)
        {
            return;
        }
        hashMap[bucket][bucketItem] = -1;
    }
}

/**
 * Your MyHashMap object will be instantiated and called as such:
 * MyHashMap obj = new MyHashMap();
 * obj.Put(key,value);
 * int param_2 = obj.Get(key);
 * obj.Remove(key);
 */